### PR TITLE
Fix opening console logs in a new window or refreshing

### LINKED ISSUE #2460: Refreshing or opening console logs in a new tab loses selected resource
Repo 1:

1. Go to resources page.
2. Open console logs for a resource in a new tab
3. Console logs page opens without resource selected

Repo 2:

1. Go to console logs page
2. Select resource
3. Refresh
4. Console logs page refreshed without resource selected

The problem is resources are fetched from DCP again when refreshing the page (new circuit) and the console logs page isn't waiting for the selected resource to be available so clears the selection.

### PR DESCRIPTION
**Pull request by JamesNK**
Fixes https://github.com/dotnet/aspire/issues/2460
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/2463)

### CODE REVIEW

File: src/Aspire.Dashboard/Components/Pages/ConsoleLogs.razor.cs [hunk 1]
```diff
@@ -85,18 +119,27 @@ await foreach (var (changeType, resource) in subscription.WithCancellation(_reso
...
+            await SetSelectedConsoleResource(resource);
+
+            Logger.LogDebug("Selected console resource from name {ResourceName}.", ResourceName);
+            loadingTcs.TrySetResult();
+        }
     }
 
     protected override async Task OnParametersSetAsync()
     {
+        Logger.LogDebug($"Initializing console logs view model.");
```
Comment (davidfowl) [inline src/Aspire.Dashboard/Components/Pages/ConsoleLogs.razor.cs:142]:
```suggestion
        Logger.LogDebug("Initializing console logs view model.");
```

**Comment by JamesNK** [comment]
@joperezr @davidfowl Bug fix I'd like for P4 backport.

**Comment by JamesNK** [comment]
Someone review please 🙏

File: src/Aspire.Dashboard/Components/Pages/ConsoleLogs.razor.cs [hunk 2]
```diff
@@ -48,14 +53,27 @@ public sealed partial class ConsoleLogs : ComponentBase, IAsyncDisposable, IPage
...
     public string SessionStorageKey => "ConsoleLogs_PageState";
 
-    protected override void OnInitialized()
+    protected override async Task OnInitializedAsync()
     {
         _noSelection = new() { Id = null, Name = ControlsStringsLoc[nameof(ControlsStrings.SelectAResource)] };
         PageViewModel = new ConsoleLogsViewModel { SelectedOption = _noSelection, SelectedResource = null, Status = Loc[nameof(Dashboard.Resources.ConsoleLogs.ConsoleLogsLoadingResources)] };
 
-        TrackResourceSnapshots();
+        var loadingTcs = new TaskCompletionSource();
```
Comment (davidfowl) [inline src/Aspire.Dashboard/Components/Pages/ConsoleLogs.razor.cs:61]:
```suggestion
        var loadingTcs = new TaskCompletionSource(TaskCreationOptions.RunContinuationsAsynchronously);
```

**Review by tlmii — APPROVED** [review]

**Review by davidfowl — APPROVED** [review]

**Comment by joperezr** [comment]
Oops, you and I called the backport at the same time, I will cancel one of those

**Pull request merged**